runtime/src/main/java/io/micronaut/logging/LogbackLogLevelConfigurer.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.logging;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.env.Environment;
+import io.micronaut.context.event.ApplicationEventListener;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.naming.conventions.StringConvention;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.runtime.context.scope.refresh.RefreshEvent;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+
+/**
+ * Logback log level configurer.
+ *
+ * @author Denis Stepanov
+ * @since 1.3.0
+ */
+@BootstrapContextCompatible
+@Singleton
+@Context
+@Requires(classes = ch.qos.logback.classic.LoggerContext.class)
+@Requires(beans = Environment.class)
+@Requires(property = LogbackLogLevelConfigurer.LOGGER_LEVELS_PROPERTY_PREFIX)
+@Internal
+final class LogbackLogLevelConfigurer implements ApplicationEventListener<RefreshEvent> {
+
+    static final String LOGGER_LEVELS_PROPERTY_PREFIX = "logger.levels";
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(LogbackLogLevelConfigurer.class);
+
+    private final Environment environment;
+
+    /**
+     * Sets log level according to properties.
+     *
+     * @param environment The  environment
+     */
+    public LogbackLogLevelConfigurer(Environment environment) {
+        this.environment = environment;
+        configureLogLevels(environment);
+    }
+
+    /**
+     * Sets log level according to properties on refresh.
+     *
+     * @param event refresh event
+     */
+    @Override
+    public void onApplicationEvent(RefreshEvent event) {
+        configureLogLevels(environment);
```

Comment:
In this case it should only call `configureLogLevel` if `event.getSource()` contains a key that starts with `LOGGER_LEVELS_PROPERTY_PREFIX`

Reply:
done